jsonmodels/fields.py

```diff
@@ -651,6 +648,14 @@ def to_struct(self, value):
             return value.strftime(self.str_format)
         return value.isoformat()
 
+    def toBsonEncodable(self, value: datetime) -> datetime:
+        """
+        Keep datetime object a datetime object, since pymongo supports that.
+        """
+        if not isinstance(value, self.types):
+            raise ModelValidationError.invalidType(self.types, field_name=self.name)
```

Comment:
ModelValidationError is a stardust error, jsonmodels doesn't know about it. We should raise a more generic error here.